common/scala/src/main/scala/org/apache/openwhisk/core/database/elasticsearch/ElasticSearchActivationStore.scala

```diff
@@ -0,0 +1,482 @@
+            s"[PUT] 'activations' failed to put document: '${activation.docid}'; ${e.getMessage}.",
+            ErrorLevel)
+          throw PutException("error on 'put'")
+      }
+    } else {
+      client.execute(op).map { res =>
+        if (res.status == StatusCodes.OK.intValue || res.status == StatusCodes.Created.intValue) {
+          transid.finished(
+            this,
+            start,
+            s"[PUT] 'activations' completed activation: '${res.result.id}', response: '${activation.docinfo}'")
+          DocInfo(res.result.id)
+        } else {
+          transid.failed(
+            this,
+            start,
+            s"[PUT] 'activations' failed to put document: '${activation.docid}'; http status: '${res.status}'",
+            ErrorLevel)
+          throw new Exception("Unexpected http response code: " + res.status)
+        }
+      }
+    }
+
+    reportFailure(
+      res,
+      failure =>
+        transid
+          .failed(this, start, s"[PUT] 'activations' internal error, failure: '${failure.getMessage}'", ErrorLevel))
+  }
+
+  private def doStore(ops: Seq[IndexRequest])(
+    implicit transid: TransactionId): Future[Seq[Either[ArtifactStoreException, DocInfo]]] = {
+    val count = ops.size
+    val start = transid.started(this, LoggingMarkers.DATABASE_BULK_SAVE, s"'activations' saving $count documents")
+    val res = client
+      .execute {
+        bulk(ops)
+      }
+      .map { res =>
+        if (res.status == StatusCodes.OK.intValue || res.status == StatusCodes.Created.intValue) {
+          res.result.items.map { bulkRes =>
+            if (bulkRes.status == StatusCodes.OK.intValue || bulkRes.status == StatusCodes.Created.intValue)
+              Right(DocInfo(bulkRes.id))
+            else
+              Left(PutException(
+                s"Unexpected error: ${bulkRes.error.map(e => s"${e.`type`}:${e.reason}").getOrElse("unknown")}, code: ${bulkRes.status} on 'bulk_put'"))
+          }
+        } else {
+          transid.failed(
+            this,
+            start,
+            s"'activations' failed to put documents, http status: '${res.status}'",
+            ErrorLevel)
+          throw new Exception("Unexpected http response code: " + res.status)
+        }
+      }
+
+    reportFailure(
+      res,
+      failure =>
+        transid
+          .failed(this, start, s"[PUT] 'activations' internal error, failure: '${failure.getMessage}'", ErrorLevel))
+  }
+
+  override def get(activationId: ActivationId, context: UserContext)(
+    implicit transid: TransactionId): Future[WhiskActivation] = {
+
+    cacheLookup(CacheKey(activationId.asString), doGet(activationId))
+  }
+
+  private def doGet(activationId: ActivationId)(implicit transid: TransactionId): Future[WhiskActivation] = {
+
+    val start =
+      transid.started(this, LoggingMarkers.DATABASE_GET, s"[GET] 'activations' finding activation: '$activationId'")
+
+    val index = generateIndex(activationId.toString.split("/")(0))
```

Comment:
Better to abstract the split logic as a function to ensure that its done in consistent way